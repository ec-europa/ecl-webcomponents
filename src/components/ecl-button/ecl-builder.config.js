const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, '../../../node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-button-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-button-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-button-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-button-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
  ],
};
