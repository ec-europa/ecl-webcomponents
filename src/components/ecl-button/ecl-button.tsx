import { Component, Prop, h, Element } from '@stencil/core';

@Component({
  tag: 'ecl-button',
  styleUrls: {
    ec: './build/styles/ecl-button-ec.css',
    eu: './build/styles/ecl-button-eu.css',
  },
  shadow: false,
  scoped: true,
})

export class EclButton {
  @Element() el: HTMLElement;
  @Prop() styleClass: string = '';
  @Prop() type: string = 'primary';
  @Prop() theme: string = 'ec';

  componentDidRender() {
    if (this.el.getElementsByTagName('ecl-icon')[0] && this.el.querySelector('.ecl-icon')) {
      const slot = this.el.getElementsByTagName('ecl-icon')[0].getAttribute('slot');
      this.el.querySelector('.ecl-icon').classList.add('ecl-button__icon');
      if (slot) {
        this.el.querySelector('.ecl-icon').classList.add(`ecl-button__${slot.substring(0, 5) + '-' + slot.substring(5)}`);
      }
    }
  }

  getClass(): string {
    return [
      `ecl-button`,
      `ecl-button--${this.type}`,
      this.styleClass
    ].join(' ');
  }

  render() {
    return (
      <button class={this.getClass()}>
        <span class="ecl-button__container">
          <slot name="icon-before"></slot>
          <span class="ecl-button__label">
            <slot></slot>
          </span>
          <slot name="icon-after"></slot>
        </span>
      </button>
    )
  }
}
