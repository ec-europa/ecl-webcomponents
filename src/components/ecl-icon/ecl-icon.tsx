import { Component, Prop, h } from '@stencil/core';
import getAssetPath from "../../utils/assetPath";

@Component({
  tag: 'ecl-icon',
  styleUrls: {
    ec: './build/styles/ecl-icon-ec.css',
    eu: './build/styles/ecl-icon-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})

export class EclIcon {
  @Prop() styleClass: string = '';
  @Prop() icon: string = '';
  @Prop() size: string = 'xs';
  @Prop() color: string = '';
  @Prop() titleTag: string = '';
  @Prop({ mutable: true }) path: string;
  @Prop() transform: string = '';
  @Prop({ mutable: true }) theme: string;
  @Prop() sprite: string = '';

  componentWillLoad() {
    this.theme = document.documentElement.getAttribute('data-ecl-theme') ?? (this.theme || 'ec');
    
    if (!this.sprite) {
      this.path = getAssetPath(`./build/images/${this.theme}/icons.svg`);
    } else {
      this.path = getAssetPath(`./build/images/${this.theme}/${this.sprite}.svg`);
    }
  }

  getClass(): string {
    const styleClasses = [
      `ecl-icon`,
      `ecl-icon--${this.size}`,
      this.styleClass,
    ];
    if (this.color) {
      styleClasses.push(
        `ecl-icon--${this.color}`
      );
    }
    if (this.transform) {
      styleClasses.push(
        `ecl-icon--${this.transform}`
      );
    }

    return styleClasses.join(' ');
  }

  render() {
    return (
      <svg class={this.getClass()}>
      { this.titleTag ? <title>{this.titleTag}</title> : '' } 
        <use xlinkHref={`${this.path}#${this.icon}`}></use>
      </svg>
    )
  }
}