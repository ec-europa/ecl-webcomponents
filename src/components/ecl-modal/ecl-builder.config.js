const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, './node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  scripts: [
    {
      entry: path.resolve(nodeModules, '@ecl/vanilla-component-modal/modal.js'),
      dest: path.resolve(outputFolder, 'scripts/ecl-modal-vanilla.js'),
      options: {
        banner,
        moduleName: 'ECL',
        includePaths,
        sourceMap: false,
      },
    },
  ],
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-modal-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-modal-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-modal-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-modal-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
  ],
};
