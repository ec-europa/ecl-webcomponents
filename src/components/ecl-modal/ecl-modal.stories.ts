const getArgs = () => ({
  variant: 'default',
});

const getArgTypes = () => ({
  variant: {
    type: { name: 'select' },
    description: 'modal variant',
    options: [
      'default',
      'information',
      'success',
      'warning',
    ],
  }
});

export default {
  title: 'Components/modal',
};

const Template = args =>
  `<ecl-button
    type="button"
    item-id="modal-demo-toggle"
    theme="${args.theme}"
  >
    Open modal
  </ecl-button>
  <ecl-modal
    theme="${args.theme}"
    variant="${args.variant}"
    with-header
    with-footer
    with-body
    ecl-script
    toggle-id="modal-demo-toggle"
    close-label="Close"
  >
    <ecl-text slot="header" theme="${args.theme}" tag="span">Lorem ipsum dolor sit amet</ecl-text>
    <div slot="body">Sed quam augue, volutpat sed dapibus in, accumsan a arcu. 
      Nulla quam enim, porttitor at neque a, egestas porttitor tortor. Nam tortor sem, elementum id augue quis, posuere vestibulum dui.
      Donec id posuere libero, sit amet egestas lorem. Aliquam finibus ipsum mauris, a molestie tortor laoreet.
    </div>
    <div slot="footer">
      <ecl-button
        theme="${args.theme}"
        type="submit"
        variant="primary"
        style-class="ecl-modal__button"
      >
        Primary action
      </ecl-button>
      <ecl-button
        theme="${args.theme}"
        type="button"
        variant="secondary"
        style-class="ecl-modal__button"
      >
        Secondary action
      </ecl-button>
    </div>
  </ecl-modal>`;

export const Modal = Template.bind({});
Modal.storyName = 'default';
Modal.args = getArgs();
Modal.argTypes = getArgTypes();
