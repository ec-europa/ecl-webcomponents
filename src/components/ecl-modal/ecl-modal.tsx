import { Component, h, Prop, Element } from '@stencil/core';
import getAssetPath from "../../utils/assetPath";
declare const ECL: any;

@Component({
  tag: 'ecl-modal',
  styleUrls: {
    ec: './build/styles/ecl-modal-ec.css',
    eu: './build/styles/ecl-modal-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})
export class EclModal {
  @Element() el: HTMLElement;
  @Prop() theme: string = 'ec';
  @Prop() variant: string = 'default';
  @Prop() styleClass: string;
  @Prop() withBody: boolean = false;
  @Prop() withHeader: boolean = false;
  @Prop() eclScript: boolean = false;
  @Prop() withFooter: boolean = false;
  @Prop() toggleId: string;
  @Prop() closeLabel: string;
  @Prop() itemId: string;

  getClass(): string {
    return [
      `ecl-modal`,
      `ecl-modal--${this.variant}`,
      this.styleClass
    ].join(' ');
  }

  componentDidRender() {
    if (this.eclScript) {
      const src = getAssetPath('./build/scripts/ecl-modal-vanilla.js');
      if (document.querySelector(`script[src="${src}"]`)) {
        document.querySelector(`script[src="${src}"]`).remove();
      }
      const script = document.createElement('script');
      script.src = src;
      script.onload = () => {
        const modal = new ECL.Modal(this.el.firstElementChild);
        modal.init();
      };
      document.body.appendChild(script);
    }
  }

  render() {
    return (
      <dialog
        class={this.getClass()}
        data-ecl-modal-toggle={this.toggleId}
        id={this.itemId}
      >
        <div class="ecl-modal__container ecl-container">
          <div class="ecl-modal__content ecl-col-12 ecl-col-m-8 ecl-offset-m-2 ecl-col-l-6 ecl-offset-l-3">
            <header class="ecl-modal__header">
            { this.withHeader ?
              <div class="ecl-modal__header-content">
                <slot name="header"></slot>
              </div> : ''
            }
            <ecl-button
              type="button"
              variant="ghost"
              styleClass={`ecl-modal__close sc-ecl-modal-${this.theme}`}
              data-ecl-modal-close
            >
              {this.closeLabel}
              <ecl-icon
                icon="close-filled"
                size="s"
              ></ecl-icon>
              </ecl-button>
            </header>
          { this.withBody ?
            <div class="ecl-modal__body">
              <slot name="body"></slot>
            </div> : ''
          }
          { this.withFooter ?
            <footer class="ecl-modal__footer">
              <div class="ecl-modal__footer-content">
                <slot name="footer"></slot>
              </div>
            </footer> : ''
          }
          </div>
        </div>
      </dialog>
    );
  }
}
