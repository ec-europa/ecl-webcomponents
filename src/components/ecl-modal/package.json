{
  "name": "@ecl/web-components-modal",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "0.7.0",
  "description": "ECL Modal",
  "main": "ecl-modal.tsx",
  "private": true,
  "scripts": {
    "build:styles": "cross-env ecl-builder styles",
    "build:scripts": "cross-env ecl-builder scripts",
    "build": "run-s clean build:*",
    "clean": "rimraf build",
    "prepublish": "npm run build"
  },
  "devDependencies": {
    "@ecl/builder": "3.8.4",
    "@ecl/vanilla-component-modal": "3.8.4",
    "@ecl/vanilla-layout-grid": "3.8.4",
    "cross-env": "7.0.3"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}
