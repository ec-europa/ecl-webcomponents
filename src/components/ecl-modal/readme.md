# ecl-modal



<!-- Auto Generated Below -->


## Properties

| Property     | Attribute     | Description | Type      | Default     |
| ------------ | ------------- | ----------- | --------- | ----------- |
| `closeLabel` | `close-label` |             | `string`  | `undefined` |
| `eclScript`  | `ecl-script`  |             | `boolean` | `false`     |
| `itemId`     | `item-id`     |             | `string`  | `undefined` |
| `styleClass` | `style-class` |             | `string`  | `undefined` |
| `theme`      | `theme`       |             | `string`  | `'ec'`      |
| `toggleId`   | `toggle-id`   |             | `string`  | `undefined` |
| `variant`    | `variant`     |             | `string`  | `'default'` |
| `withBody`   | `with-body`   |             | `boolean` | `false`     |
| `withFooter` | `with-footer` |             | `boolean` | `false`     |
| `withHeader` | `with-header` |             | `boolean` | `false`     |


## Dependencies

### Depends on

- [ecl-button](../ecl-button)
- [ecl-icon](../ecl-icon)

### Graph
```mermaid
graph TD;
  ecl-modal --> ecl-button
  ecl-modal --> ecl-icon
  style ecl-modal fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
