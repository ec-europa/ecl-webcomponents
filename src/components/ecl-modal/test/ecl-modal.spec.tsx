import { newSpecPage } from '@stencil/core/testing';
import { EclModal } from '../ecl-modal';
import { EclButton } from '../../ecl-button/ecl-button';
import { EclIcon } from '../../ecl-icon/ecl-icon';

describe('ecl-modal', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclModal, EclButton, EclIcon],
      html: `<ecl-modal
        theme="ecl-modal"
        with-header
        with-footer
        with-body
        ecl-script
        toggle-id="modal-demo-toggle"
        close-label="Close"
      >
        <div slot="header">
          <ecl-icon
            icon="information"
            size="m"
            style-class="ecl-modal__icon sc-ecl-modal-ec"
          ></ecl-icon>
          <ecl-text tag="span">Lorem ipsum dolor sit amet</ecl-text>
        </div>
        <div slot="body">Sed quam augue, volutpat sed dapibus in, accumsan a arcu. 
          Nulla quam enim, porttitor at neque a, egestas porttitor tortor. Nam tortor sem, elementum id augue quis, posuere vestibulum dui.
          Donec id posuere libero, sit amet egestas lorem. Aliquam finibus ipsum mauris, a molestie tortor laoreet.
        </div>
        <div slot="footer">
          <ecl-button
            type="submit"
            variant="primary"
            style-class="ecl-modal__button"
          >
            Primary action
          </ecl-button>
          <ecl-button
            type="button"
            variant="secondary"
            style-class="ecl-modal__button"
          >
            Secondary action
          </ecl-button>
        </div>
      </ecl-modal>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
