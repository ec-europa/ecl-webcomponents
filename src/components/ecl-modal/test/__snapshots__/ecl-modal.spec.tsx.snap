// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-modal renders 1`] = `
<ecl-modal close-label="Close" ecl-script="" theme="ec" toggle-id="modal-demo-toggle" with-body="" with-body-fixed="" with-footer="" with-header="">
  <!---->
  <dialog class="ecl-modal ecl-modal--default ecl-modal--l" data-ecl-modal-toggle="modal-demo-toggle">
    <div class="ecl-container ecl-modal__container">
      <div class="ecl-col-l-8 ecl-col-m-10 ecl-modal__content">
        <header class="ecl-modal__header">
          <div class="ecl-modal__header-content">
            <div slot="header">
              <ecl-icon icon="information" size="m" style-class="ecl-modal__icon sc-ecl-modal-ec">
                <svg class="ecl-icon ecl-icon--m ecl-modal__icon sc-ecl-modal-ec">
                  <use xlink:href="/build/images/ec/icons.svg#information"></use>
                </svg>
              </ecl-icon>
              <ecl-text tag="span">
                Lorem ipsum dolor sit amet
              </ecl-text>
            </div>
          </div>
          <ecl-button data-ecl-modal-close>
            <!---->
            <button class="ecl-button ecl-button--ghost ecl-modal__close sc-ecl-modal-ec" data-ecl-modal-close="eclModalClose" type="button">
              <span class="ecl-button__container">
                <span class="ecl-button__label">
                  Close
                  <ecl-icon>
                    <svg class="ecl-button__icon ecl-icon ecl-icon--s">
                      <use xlink:href="/build/images/ec/icons.svg#close-filled"></use>
                    </svg>
                  </ecl-icon>
                </span>
              </span>
            </button>
          </ecl-button>
        </header>
        <div class="ecl-modal__body">
          <div class="ecl-modal__body-scroll" data-ecl-modal-scroll>
            <div slot="body">
              Sed quam augue, volutpat sed dapibus in, accumsan a arcu. Nulla quam enim, porttitor at neque a, egestas porttitor tortor. Nam tortor sem, elementum id augue quis, posuere vestibulum dui. Donec id posuere libero, sit amet egestas lorem. Aliquam finibus ipsum mauris, a molestie tortor laoreet.
            </div>
          </div>
          <div aria-hidden="true" class="ecl-modal__body-overflow"></div>
          <div class="ecl-modal__body-fixed">
            <div slot="body-fixed">
              <ecl-input input-class="ecl-checkbox__input" input-id="checkbox-default-1" label="Option label" name="checkbox-default" theme="ec" type="checkbox">
                <div class="ecl-checkbox ecl-checkbox--m ecl-input">
                  <input class="ecl-checkbox__input" id="checkbox-default-1" name="checkbox-default" type="checkbox">
                  <label class="ecl-checkbox__label" htmlfor="checkbox-default-1">
                    <span class="ecl-checkbox__box">
                      <ecl-icon style-class="ecl-checkbox__icon sc-ecl-input-ec">
                        <svg class="ecl-checkbox__icon ecl-icon ecl-icon--m sc-ecl-input-ec">
                          <use xlink:href="/build/images/ec/icons.svg#check"></use>
                        </svg>
                      </ecl-icon>
                    </span>
                    <span class="ecl-checkbox__text">
                      Option label
                    </span>
                  </label>
                </div>
              </ecl-input>
            </div>
          </div>
        </div>
        <footer class="ecl-modal__footer">
          <div class="ecl-modal__footer-content">
            <div slot="footer">
              <ecl-button style-class="ecl-modal__button" type="button" variant="secondary">
                <!---->
                <button class="ecl-button ecl-button--secondary ecl-modal__button" type="button">
                  <span class="ecl-button__container">
                    <span class="ecl-button__label">
                      Secondary action
                    </span>
                  </span>
                </button>
              </ecl-button>
              <ecl-button style-class="ecl-modal__button" type="submit" variant="primary">
                <!---->
                <button class="ecl-button ecl-button--primary ecl-modal__button" type="submit">
                  <span class="ecl-button__container">
                    <span class="ecl-button__label">
                      Primary action
                    </span>
                  </span>
                </button>
              </ecl-button>
            </div>
          </div>
        </footer>
      </div>
    </div>
  </dialog>
</ecl-modal>
`;
